crates/holochain_websocket/src/lib.rs

```diff
@@ -660,11 +662,50 @@ impl ConnectRequest {
     }
 }
 
+// TODO async_trait still needed for dynamic dispatch https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#dynamic-dispatch
+#[async_trait::async_trait]
+trait TcpListener: Send + Sync {
+    async fn accept(&self) -> Result<(tokio::net::TcpStream, SocketAddr)>;
+
+    fn local_addrs(&self) -> Result<Vec<SocketAddr>>;
+}
+
+#[async_trait::async_trait]
+impl TcpListener for tokio::net::TcpListener {
+    async fn accept(&self) -> Result<(tokio::net::TcpStream, SocketAddr)> {
+        self.accept().await
+    }
+
+    fn local_addrs(&self) -> Result<Vec<SocketAddr>> {
+        Ok(vec![self.local_addr()?])
+    }
+}
+
+struct DualStackListener {
+    v4: tokio::net::TcpListener,
+    v6: tokio::net::TcpListener,
+}
+
+#[async_trait::async_trait]
+impl TcpListener for DualStackListener {
+    async fn accept(&self) -> Result<(tokio::net::TcpStream, SocketAddr)> {
+        let (stream, addr) = select! {
```

Comment:
`accept` is documented to be cancel safe so that it is acceptable to use it within a `select!`